astropy/table/tests/test_table.py

```diff
@@ -2359,36 +2349,34 @@ def test_replace_update_column_via_setitem_warnings_refcount():
     t = table.Table([[1, 2, 3], [4, 5, 6]], names=['a', 'b'])
     ta = t['a']  # noqa : Generate an extra reference to original column
 
-    with catch_warnings() as w:
+    with pytest.warns(TableReplaceWarning, match="replaced column 'a' and the "
+                      "number of references") as w:
         with table.conf.set_temp('replace_warnings',
                                  ['refcount', 'attributes', 'slice']):
             t['a'] = [10, 20, 30]
-        assert len(w) == 1
-        assert "replaced column 'a' and the number of references" in str(w[0].message)
+    assert len(w) == 1
 
 
 def test_replace_update_column_via_setitem_warnings_always():
     """
     Test warnings related to table replace change in #5556:
     Test 'always' setting that raises warning for any replace.
     """
+    from inspect import currentframe, getframeinfo
+
     t = table.Table([[1, 2, 3], [4, 5, 6]], names=['a', 'b'])
 
-    with catch_warnings() as w:
-        with table.conf.set_temp('replace_warnings', ['always']):
-            t['a'] = 0  # in-place slice update
-            assert len(w) == 0
+    with table.conf.set_temp('replace_warnings', ['always']):
+        t['a'] = 0  # in-place slice update
 
-            from inspect import currentframe, getframeinfo
-            frameinfo = getframeinfo(currentframe())
+        frameinfo = getframeinfo(currentframe())
+        with pytest.warns(TableReplaceWarning, match="replaced column 'a'") as w:
             t['a'] = [10, 20, 30]  # replace column
-            assert len(w) == 1
-            assert "replaced column 'a'" == str(w[0].message)
+        assert len(w) == 1
 
-            # Make sure the warning points back to the user code line
-            assert w[0].lineno == frameinfo.lineno + 1
-            assert w[0].category is table.TableReplaceWarning
-            assert 'test_table' in w[0].filename
+        # Make sure the warning points back to the user code line
+        assert w[0].lineno == frameinfo.lineno + 2
```

Comment:
I have no idea why it changed from `+1` to `+2`. Maybe because I moved some orders around? Is this okay or do I have to make it `+1` again?